src/Microsoft.Tye.Core/ApplicationFactory.cs

```diff
@@ -398,14 +398,10 @@ private static void AddToRootServices(ApplicationBuilder root, HashSet<string> d
             // Add ourselves in the set of all current dependencies.
             dependencies.Add(serviceName);
 
-            // Iterate through all services and add the current services as a dependency (except ourselves)
+            // Iterate through all services and add the current services as a dependency
             foreach (var s in root.Services)
             {
-                if (dependencies.Contains(s.Name, StringComparer.OrdinalIgnoreCase)
-                    && !s.Name.Equals(serviceName, StringComparison.OrdinalIgnoreCase))
-                {
-                    s.Dependencies.Add(serviceName);
-                }
+                s.Dependencies.Add(serviceName);
```

Comment:
does this mean that every service has a dependency on itself? 

If so, why isn't the right fix elsewhere (use env-vars from service's dependencies + the service)?

Reply:
Yeah, you are right that fix is more sensible.